perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import warnings
+
+from perceval.runtime import RemoteJob
+from perceval.runtime.rpc_handler import RPCHandler
+from perceval.utils import PersistentData, FileFormat
+from perceval.utils.logging import get_logger, channel
+import os
+import json
+import datetime
+
+FILE_EXT_JGRP = 'jgrp'
+MAX_SAMPLE_DEF_GROUP = 10000
+
+class JobGroup:
+    """
+    A JobGroup handles a collection of Jobs saved on disk using perceval
+    persistent data. It can perform various tasks such as
+    - Saving information for a collection of jobs, whether they have been sent to the cloud or not.
+    - Running jobs within the group either in parallel or sequentially.
+    - Rerunning failed jobs within the group.
+    """
+    def __init__(self, name: str):
+        self._name = name
+        self._group_info = dict()
```

Comment:
It's better to store a `PersistentData` instance in your class (it'll help doing the mocks for unit tests at least) Check how it's done in `Config` or `TokenProvider` classes for reference.